Zend/zend_execute.c

```diff
@@ -2082,9 +2120,10 @@ static zend_always_inline zval *zend_fetch_dimension_address_inner(HashTable *ht
 				retval = &EG(uninitialized_zval);
 				break;
 			case BP_VAR_RW:
-				zend_undefined_offset(hval);
-				retval = zend_hash_index_update(ht, hval, &EG(uninitialized_zval));
-				break;
+				if (UNEXPECTED(zend_undefined_offset_write(ht, hval) == FAILURE)) {
+					return NULL;
+				}
+				/* break missing intentionally */
```

Comment:
I'm using add_new instead of update here now, because there should no longer be any way for the missing key to be created while the notice is thrown (as we incremented refcount, so the array would get separated if it's modified).